src/meta/src/manager/catalog/mod.rs

```diff
@@ -867,6 +895,21 @@ impl CatalogManager {
             database_core.increase_relation_ref_count(dependent_relation_id);
         }
         user_core.increase_ref(table.owner);
+        let _version = self
```

Comment:
Is this the only difference? If so, can we dispatch by the `table_type` instead of duplication? Also, what about leaving some comments here showing the reason for varying behaviors?

Reply:
Besides that, `TABLE` metadata doesn't need to be committed here, whereas MV's metadata does.

Prefer to just duplicate it, further considering the different functionality between both.

It's easier to understand what's specific to MVs and what's specific to tables, rather than coupling them in a single function.

Reply:
Oh and I suppose we can get rid of `internal` table logic. That's not needed by tables at all.